src/utils/server.cr

```diff
@@ -12,17 +12,45 @@ module Mint
       end
     end
 
-    def run
+    def port_open?(ip, port)
+      TCPSocket.new(ip, port).close
+      false
+    rescue
+      true
+    end
+
+    def run(port : Int32 = 3000)
       config = Kemal.config
       config.logger = Logger.new
       config.setup
 
-      server =
-        HTTP::Server.new(config.host_binding, config.port, config.handlers)
+      if port_open?(config.host_binding, port)
+        server =
+          HTTP::Server.new(config.host_binding, port, config.handlers)
+        terminal.print "#{COG} Development server started on http://#{config.host_binding}:#{port}/\n"
+      else
```

Comment:
I think it should fail in this case. Having blocking terminal IO on a webserver isn't good practice since it looks like the server is running while in reality it's waiting for user input.
So we could at least check if the terminal is a tty first and fail otherwise. Something like `STDIN.tty?` should do the trick.

Reply:
Yes @manveru is right, someone might use `mint start` as production server for whatever reason and suppressing the output like `mint start > /dev/null` would cause it to behave like it's running while in reality it's waiting for user input. 

This should be the branch of `elseif STDIN.tty?` and then the else branch would just exit with a nice message and a code of `1`.

Reply:
Now I understand what are you talking about, going to add this today.